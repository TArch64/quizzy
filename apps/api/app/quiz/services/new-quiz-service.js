export class NewQuizService {
    #prisma;

    constructor({ prisma }) {
        this.#prisma = prisma;
    }

    create(data) {
        return this.#prisma.quiz.create({
            data: {
                questions: {
                    create: data.questions.map(this.#formatQuestionRecord.bind(this))
                }
            }
        });
    }

    #formatQuestionRecord(data) {
        return {
            id: data.id,
            question: data.question,
            correct: data.correct,
            answers: {
                create: data.answers.map(this.#formatAnswerRecord.bind(this))
            }
        }
    }

    #formatAnswerRecord(data) {
        return {
            id: data.id,
            text: data.text
        };
    }
}
