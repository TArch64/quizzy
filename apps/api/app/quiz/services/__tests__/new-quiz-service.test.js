import {NewQuizService} from "../new-quiz-service.js";

const createPrisma = () => ({
    quiz: {
        create: jest.fn().mockResolvedValue()
    }
});

function createService({ prisma }) {
    return new NewQuizService({ prisma });
}

describe('create quiz', () => {
    test('should format db payload', async () => {
        const prisma = createPrisma();
        const service = createService({ prisma });

        await service.create({
            questions: [
                {
                    id: 1,
                    text: 'first-question',
                    correctId: 'first-answer',
                    answers: [
                        {
                            id: 'first-answer',
                            text: 'first'
                        },
                        {
                            id: 'second-answer',
                            text: 'second'
                        }
                    ]
                }
            ]
        });

        expect(prisma.quiz.create.mock.lastCall).toMatchSnapshot();
    });
});
