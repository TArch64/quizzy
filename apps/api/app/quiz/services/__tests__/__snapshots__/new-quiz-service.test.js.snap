// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create quiz should format db payload 1`] = `
[
  {
    "data": {
      "questions": {
        "create": [
          {
            "answers": {
              "create": [
                {
                  "id": "first-answer",
                  "text": "first",
                },
                {
                  "id": "second-answer",
                  "text": "second",
                },
              ],
            },
            "correctId": "first-answer",
            "id": 1,
            "text": "first-question",
          },
        ],
      },
    },
  },
]
`;
